docs/source/running-jobs/priorities.rst

```diff
@@ -0,0 +1,182 @@
+.. _job-priorities:
+
+Managed Jobs with Priorities
+============================
+
+SkyPilot supports priority-based scheduling, preemption, and re-queuing of jobs running on Kubernetes. You can achieve this by leveraging Kubernetes' native priority classes.
+
+.. tip::
+   Jobs with priorities and preemption are only supported on Kubernetes.
+
+To set job priorities:
+
+1. Create :ref:`priority classes <priorities-example-priority-classes>` in your Kubernetes cluster.
+2. :ref:`Set the priority classes in your SkyPilot jobs<priorities-example-sky-pilot-jobs>` by setting ``experimental.config_overrides.kubernetes.pod_config.spec.priorityClassName``.
+3. Use :ref:`sky jobs launch <managed-jobs>` to launch your jobs.
+
+With this setup, you can run high priority jobs that preempt low priority jobs when resources are constrained.
+
+How Priorities and Preemptions Work
+-----------------------------------
+
+When the cluster does not have enough resources to run all jobs, **high priority jobs will preempt low priority jobs.** This means pods of low priority jobs will be terminated to create space for high priority jobs.
+
+Preempted jobs will be automatically rescheduled by SkyPilot when resources become available again. You can set up :ref:`checkpointing and recovery <checkpointing>` in your code to reduce wasted work.
+
+Jobs with the same priority level follow SkyPilot's :ref:`default scheduling behavior <job-queue>`.
+
+.. tip::
+   You can also apply priority classes to unmanaged SkyPilot clusters. However, when unmanaged clusters are preempted, they will not be automatically restarted.
+
+Limitations
+-----------
+
+..
+  TODO: The first point is a very severe limitation. We need to fix this and do a better job of explaining it.
+
+1. When the job queue is longer than 4x the number of CPUs on the SkyPilot controller, any new high priority jobs will not be able to preempt low priority jobs until they are at the front of the queue.
```

Comment:
notes from this morning: Don't need to say `4x the number of CPUs`, instead just indicate that there is some limit (e.g. "When the job queue is too long, ...")

Reply:
Updated to remove the limitation for now